subprojects/language-jvm/src/main/java/org/gradle/api/tasks/compile/AbstractOptions.java

```diff
@@ -42,52 +37,8 @@ public void define(@Nullable Map<String, Object> args) {
         }
     }
 
-    public Map<String, Object> optionMap() {
```

Comment:
It would be nice to get rid of the `define` method as well and then remove the whole class.

We could at least deprecate the methods which use the `define` method: `CompileOptions.fork`, `CompileOptions.debug` and `GroovyCompileOptions.fork`. WDYT?